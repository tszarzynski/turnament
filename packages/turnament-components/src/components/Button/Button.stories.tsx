import React from "react";
import Button from "./Button";
import IconNext from "../IconNext";

export default {
	component: Button,
	title: "Button",
};

export const Default = () => {
	return <Button variant="primary">Click me</Button>;
};

export const Secondary = () => {
	return <Button variant="secondary">Click me</Button>;
};

export const Disabled = () => {
	return <Button disabled={true}>Click me</Button>;
};

export const WithIcon = () => {
	return (
		<Button fullWidth={true} iconSlot={<IconNext />}>
			Click me
		</Button>
	);
};

export const SecondaryWithIcon = () => {
	return (
		<Button variant="secondary" fullWidth={true} iconSlot={<IconNext />}>
			Click me
		</Button>
	);
};
