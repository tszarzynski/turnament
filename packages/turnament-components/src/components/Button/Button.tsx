import type React from "react";

type Props = {
	children: React.ReactNode;
	variant?: "primary" | "secondary";
	disabled?: boolean;
	type?: "button" | "submit" | "reset";
	className?: string;
	onClick?: () => void;
	fullWidth?: boolean;
	iconSlot?: React.ReactNode;
	title?: string;
};

const Button = ({
	children,
	variant = "primary",
	disabled = false,
	type = "button",
	className = "",
	onClick,
	fullWidth = false,
	iconSlot,
	title,
}: Props) => {
	// Base button styles
	const baseStyles =
		"font-medium text-md uppercase hover:opacity-50 transition-opacity  focus:outline-none focus:ring-2 focus:ring-offset-1 whitespace-nowrap";

	// Variant-specific styles
	const variantStyles = {
		primary: "cursor-pointer border-primary text-primary focus:ring-red-400",
		secondary:
			"cursor-pointer border-secondary text-secondary  focus:ring-blue-400",
	};

	// Disabled styles override variant styles
	const disabledStyles =
		"border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent";

	// Full width style
	const widthStyle = fullWidth ? "w-full" : "";

	return (
		<button
			type={type}
			disabled={disabled}
			className={`
        ${baseStyles}
        ${disabled ? disabledStyles : variantStyles[variant]}
        ${widthStyle}
        ${className}
      `}
			onClick={disabled ? undefined : onClick}
			title={title}
		>
			<div className="flex flex-row items-stretch justify-between w-full border-1">
				<span className="flex items-center flex-1 px-4 py-2 ">{children}</span>
				{iconSlot && (
					<span className="border-l-1">
						<i className="h-10 w-10 flex items-center">{iconSlot}</i>
					</span>
				)}
			</div>
		</button>
	);
};

export default Button;
