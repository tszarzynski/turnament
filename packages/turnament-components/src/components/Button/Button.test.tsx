import { render } from "vitest-browser-react";
import Button from "./Button";

describe("Button", () => {
	it("renders without error", () => {
		const { asFragment } = render(<Button>Click me</Button>);
		expect(asFragment()).toMatchSnapshot();
	});
});
