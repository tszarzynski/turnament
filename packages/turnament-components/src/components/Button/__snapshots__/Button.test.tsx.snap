// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders without error 1`] = `
<DocumentFragment>
  <button
    class="
        font-medium text-lg uppercase hover:opacity-50 transition-opacity  focus:outline-none focus:ring-2 focus:ring-offset-1 whitespace-nowrap
        cursor-pointer border-primary text-primary focus:ring-red-400
        
        
      "
    type="button"
  >
    <div
      class="flex flex-row items-stretch justify-between w-full border-1"
    >
      <span
        class="flex items-center flex-1 px-4 py-2 "
      >
        Click me
      </span>
    </div>
  </button>
</DocumentFragment>
`;
